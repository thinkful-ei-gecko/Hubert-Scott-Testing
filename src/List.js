import React from 'react';
import ReactDOM from 'react-dom';
import Card from './Card';
import './index.css';

function List(props){

    const title = props.cards[0].title;
    const content = props.cards[0].content;

    return (
        <section className="List">
            <header className="List-header">
                <h2>{props.header}</h2>
            </header>

            <div className='List-cards'>
                <Card
                title = {title}
                content = {content}
                ></Card>

            </div>

         {/* <div className="List-cards">
            {props.cards.map((card) => {
                return <Card 
                key = {card.id}
                title = {card.title}
                content = {card.content} />
            })}
            <button type="button" className="List-button">
                delete
            </button>
         </ div> */}
        </ section>     
    )
}

export default List;

