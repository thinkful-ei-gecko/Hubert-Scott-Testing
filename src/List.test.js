import React from 'react';
import ReactDOM from 'react-dom';
import List from './List';
import renderer from 'react-test-renderer';

describe('<List />', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');

        ReactDOM.render(<List header='Trelloyes' cards={[{id: 'a', title: 'test snapshot', content: 'content' }, {id: 'b', title: 'test snapshot 2', content: 'more content'}]} />, div)

        ReactDOM.unmountComponentAtNode(div);
    });
});

describe('<List />', () => {
    it('renders UI correctly', () => {
        const tree = renderer.create(<List header="test header" cards={[{id: 'a', title: 'test snapshot', content: 'content' }, {id: 'b', title: 'test snapshot 2', content: 'more content'}]} />).toJSON();
        expect(tree).toMatchSnapshot();
    })
})