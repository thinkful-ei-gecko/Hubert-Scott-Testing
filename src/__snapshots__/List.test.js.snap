// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> renders UI correctly 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      test header
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        test snapshot
      </h3>
      <p>
        content
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        test snapshot 2
      </h3>
      <p>
        more content
      </p>
    </div>
    <button
      className="List-button"
      type="button"
    >
      delete
    </button>
  </div>
</section>
`;
